yarn-project/p2p/src/client/p2p_client.ts

```diff
@@ -332,7 +332,18 @@ export class P2PClient<T extends P2PClientType = P2PClientType.Full>
    * Uses the batched Request Response protocol to request a set of transactions from the network.
    */
   public async requestTxsByHash(txHashes: TxHash[]): Promise<(Tx | undefined)[]> {
-    const txs = await this.p2pService.sendBatchRequest(ReqRespSubProtocol.TX, txHashes);
+    // Set collective timeout based on the number of txs we are requesting
+    const timeoutMs = Math.max(5000, txHashes.length * 100);
+    const maxPeers = Math.ceil(txHashes.length / 3);
+    const maxRetryAttempts = Math.min(txHashes.length / maxPeers, 5);
```

Comment:
I may be missing something, but isn't this always equal to 3, since `maxPeers` is already defined based on `txHashes.length`?